languages/wikipron/src/scrape.py

```diff
@@ -64,101 +29,109 @@ def _call_scrape(lang, config, filename_suffix):
-        ]
-        phonemic_config = wikipron.Config(
-            key=iso639_code,
-            casefold=languages[iso639_code]["casefold"],
-            no_stress=True,
-            no_syllable_boundaries=True,
-        )
-        phonemic_count = _call_scrape(
-            iso639_code, phonemic_config, "_phonemic"
+def build_config_and_filter_scrape_results(
+    config_settings, wiki_name,
+    dialect_suffix=''
+):
+    path_affix = f'../tsv/{config_settings["key"]}_{dialect_suffix}'
+
+    phonemic_config = wikipron.Config(
+        **config_settings,
+    )
+    phonemic_path = f"{path_affix}phonemic.tsv"
+    phonemic_count = _call_scrape(
+        config_settings["key"], phonemic_config, phonemic_path
+    )
+
+    phonetic_config = wikipron.Config(
+        phonetic=True,
+        **config_settings,
+    )
+    phonetic_path = f"{path_affix}phonetic.tsv"
+    # Skips phonetic if phonemic failed to complete scrape.
+    if phonemic_count is not None:
+        phonetic_count = _call_scrape(
+            config_settings["key"], phonetic_config, phonetic_path
         )
-        phonetic_config = wikipron.Config(
-            key=iso639_code,
-            casefold=languages[iso639_code]["casefold"],
-            phonetic=True,
-            no_stress=True,
-            no_syllable_boundaries=True,
+    # Remove files for languages that failed to be scraped
+    # within set amount of retries.
+    if phonemic_count is None or phonetic_count is None:
+        logging.info(
+            'Failed to scrape "%s", moving on to next language. %s',
+            wiki_name,
+            {config_settings["key"]: config_settings},
         )
-        # Skips phonetic if phonemic failed to complete scrape.
-        if phonemic_count is not None:
-            phonetic_count = _call_scrape(
-                iso639_code, phonetic_config, "_phonetic"
-            )
-        phonemic_path = f"../tsv/{iso639_code}_phonemic.tsv"
-        phonetic_path = f"../tsv/{iso639_code}_phonetic.tsv"
-        # Remove files for languages that failed to be scraped
-        # within set amount of retries.
-        if phonemic_count is None or phonetic_count is None:
-            logging.info(
-                'Failed to scrape "%s", moving on to next language.',
-                languages[iso639_code]["wiktionary_name"],
-            )
-            if os.path.exists(phonemic_path):
-                os.remove(phonemic_path)
-            if os.path.exists(phonetic_path):
-                os.remove(phonetic_path)
-            continue
-        # Remove files for languages that returned nothing
-        elif phonemic_count == 0 and phonetic_count == 0:
-            logging.info(
-                '"%s", returned no entries in phonemic and phonetic settings.',
-                languages[iso639_code]["wiktionary_name"],
-            )
+        if os.path.exists(phonemic_path):
             os.remove(phonemic_path)
+        if os.path.exists(phonetic_path):
```

Comment:
Just noticed this. This is a "dark pattern" because it is theoretically possible for the file to blink in or out of existence between the check for existence and the remove. The recommended pattern is instead to do a `try: os.remove(...)` and then catch and ignore the exception (`OSError` probably) if it fails. I realize this is an annoying one...

Reply:
Found this from googling:
```
from contextlib import suppress

with suppress(OSError):
    os.remove(filename)
```
Seems nicer than ignoring the exception. Let me know if you have a preference. Is it the `os.path.exists()` check that makes the file potentially blink in or out of existence? Do I need to add this to all `os.remove()` statements or just the ones that involve `os.path.exists()`